### PR TITLE
Fix `wrappers.Request.application` docs

### PR DESCRIPTION
**Pull request by wchargin**
The request is given as the last argument, not the first.

wchargin-branch: wrappers-application-docs

### CODE REVIEW

**Comment by davidism** [comment]
This is sort of misleading too, as all "last" really means in this case is "after self if the application is a method", but in effect that's the first real argument.

**Comment by wchargin** [comment]
But it’s not just “after `self`”; it can be used with
`functools.partial`:

```
$ cat /tmp/test.py
from __future__ import print_function

import functools
import werkzeug


@werkzeug.wrappers.Request.application
def foo(x, y, request):
  return werkzeug.Response("{} + {} = {}".format(x, y, x + y))

foo_one_two = functools.partial(foo, 1, 2)

print(list(foo_one_two({"REQUEST_METHOD": "GET"}, print)))
$ python /tmp/test.py
200 OK [('Content-Type', 'text/plain; charset=utf-8'), ('Content-Length', '9')]
['1 + 2 = 3']
```

If I’m reading the code correctly—

```python
resp = f(*args[:-2] + (request,))
```

—it looks pretty clear that `request` is actually the last argument, and
that any arguments prior to the WSGI `environ` and `start_response` are
passed through to the underlying function, which makes _both_ `self` and
other bound arguments work.

File: src/werkzeug/wrappers/base_request.py [hunk 1]
```diff
@@ -205,9 +205,9 @@ def from_values(cls, *args, **kwargs):
 
     @classmethod
     def application(cls, f):
-        """Decorate a function as responder that accepts the request as first
+        """Decorate a function as responder that accepts the request as last
```
Comment (davidism) [inline src/werkzeug/wrappers/base_request.py]: "the last"

**Comment by davidism** [comment]
Ah, you're right, the change is fine then.

File: src/werkzeug/wrappers/base_request.py (hunk 1 above)
Comment (wchargin) [reply src/werkzeug/wrappers/base_request.py]: Done.

**Comment by wchargin** [comment]
Thank you!

**Pull request merged**